dialoguekit/core/dialogue.py

```diff
@@ -9,15 +9,24 @@
 
 
 class Dialogue:
-    def __init__(self, agent_id: str, user_id: str) -> None:
+    def __init__(
+        self, agent_id: str, user_id: str, conversation_id: str = None
+    ) -> None:
         """Represents a dialogue.
 
         Args:
             agent_id: Agent ID.
             user_id: User ID.
+            conversation_id: Conversation ID. Defaults to None.
         """
         self._agent_id = agent_id
         self._user_id = user_id
+        if conversation_id is None:
+            date = datetime.datetime.utcnow()
+            utc_time = calendar.timegm(date.utctimetuple())
+            self._conversation_id = str(utc_time)
```

Comment:
Note: this might not be unique. You could perhaps use f"{self._agent_id}-{self._user_id}-{str(utr_time)}"